cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -3112,7 +3138,31 @@ writeFileTextEnvelope path mbDescr a =
     runExceptT $ do
       handleIOExceptT (FileIOError path) $ BS.writeFile path content
   where
-    content = LBS.toStrict $ encodePretty' TextView.textViewJSONConfig (serialiseToTextEnvelope mbDescr a) <> "\n"
+    content = LBS.toStrict $ textEnvelopeToJSON mbDescr a
+
+writeTextEnvelopeFileWithOwnerPermissions
```

Comment:
Can we do what I suggested before and split this into a generic `writeFileWithOwnerPermissions` and the specific `writeFileTextEnvelopeWithOwnerPermissions`. That'll make it clearer what is going on, and improve opportunities for reuse.

Reply:
Sorry! Lost track of things when investigating the windows permissions.